Xamarin.Forms.Platform.Android/AppCompat/MasterDetailContainer.cs

```diff
@@ -106,6 +107,37 @@ protected override void AddChildView(VisualElement childView)
 			}
 		}
 
+		protected override void Dispose(bool disposing)
+		{
+			if (_disposed)
+			{
+				return;
+			}
+
+			_disposed = true;
```

Comment:
You're right. I wasn't thinking straight. The first time I looked at the style of your dispose, it threw me off because it wasn't familiar. I also agree about double dispose.